api.go

```diff
@@ -1727,3 +1727,678 @@ func (sc *Client) GetTimeseries(corpName, siteName string, query url.Values) ([]
 
 	return t.Data, nil
 }
+
+// CreateSiteBody is the structure required to create a Site.
+type CreateSiteBody struct {
+	Name                 string `json:"name,omitempty"`          //Identifying name of the site
+	DisplayName          string `json:"displayName,omitempty"`   //Display name of the site
+	AgentLevel           string `json:"agentLevel,omitempty"`    //Agent action level - 'block', 'log' or 'off'
+	AgentAnonMode        string `json:"agentAnonMode,omitempty"` //Agent IP anonimization mode - 'EU' or 'off'
+	BlockHTTPCode        int    `json:"blockHTTPCode,omitempty"` //HTTP response code to send when when traffic is being blocked
+	BlockDurationSeconds int    `json:"blockDurationSeconds"`    //Duration to block an IP in seconds
+}
+
+// CreateSite Creates a site in a corp.
+func (sc Client) CreateSite(corpName string, body CreateSiteBody) (Site, error) {
+	b, err := json.Marshal(body)
+	if err != nil {
+		return Site{}, logError(1, fmt.Errorf("%s with input %#v", err.Error(), body))
+	}
+
+	resp, err := sc.doRequest("POST", fmt.Sprintf("/v0/corps/%s/sites", corpName), string(b))
+	if err != nil {
+		return Site{}, logError(1, fmt.Errorf("%s with input %v", err.Error(), string(b)))
+	}
+
+	var site Site
+	err = json.Unmarshal(resp, &site)
+	if err != nil {
+		return Site{}, logError(1, fmt.Errorf("%s with input %v", err.Error(), resp))
+	}
+	return site, nil
+}
+
+// DeleteSite deltes the site
+func (sc Client) DeleteSite(corpName, siteName string) error {
+	_, err := sc.doRequest("DELETE", fmt.Sprintf("/v0/corps/%s/sites/%s", corpName, siteName), "")
+
+	if err != nil {
+		return logError(1, fmt.Errorf("%s could not delete %s in %s ", err.Error(), corpName, siteName))
+	}
+	return nil
+}
+
+// Condition contains rule condition
+type Condition struct {
+	Type          string      `json:"type,omitempty"`          //(group, single)
+	GroupOperator string      `json:"groupOperator,omitempty"` //type: group - Conditions that must be matched when evaluating the request (all, any)
+	Field         string      `json:"field,omitempty"`         //type: single - (scheme, method, path, useragent, domain, ip, responseCode, agentname, paramname, paramvalue, country, name, valueString, valueIp, signalType)
+	Operator      string      `json:"operator,omitempty"`      //type: single - (equals, doesNotEqual, contains, doesNotContain, like, notLike, exists, doesNotExist, inList, notInList)
+	Value         string      `json:"value,omitempty"`         //type: single - See request fields (https://docs.signalsciences.net/using-signal-sciences/features/rules/#request-fields)
+	Conditions    []Condition `json:"conditions,omitempty"`
+}
+
+// Action contains the rule action
+type Action struct {
+	Type string `json:"type,omitempty"` //(block, allow, exclude)
+}
+
+//CreateSiteRuleBody contains the rule for the site
+type CreateSiteRuleBody struct {
+	Type          string      `json:"type,omitempty,omitempty"` //(group, single)
```

Comment:
`Type` here is `signal|request|rateLimit`, not group or single.
Also, this structure is missing some fields for rate limiting.